source/extensions/filters/http/grpc_stats/grpc_stats_filter.cc

```diff
@@ -152,7 +155,12 @@ class GrpcStatsFilter : public Http::PassThroughFilter {
       if (cluster_) {
         if (config_->stats_for_all_methods_) {
           // Get dynamically-allocated Context::RequestStatNames from the context.
-          request_names_ = config_->context_.resolveDynamicServiceAndMethod(headers.Path());
+          if (config_->replace_dots_in_grpc_service_name_) {
+            request_names_ =
+                config_->context_.resolveDynamicServiceAndMethodWithDotReplaced(headers.Path());
+          } else {
+            request_names_ = config_->context_.resolveDynamicServiceAndMethod(headers.Path());
```

Comment:
resolveDynamicServiceAndMethodWithDotReplaced gives a impression that dot will be replaced from both the method and the service name. But actually, it is replaced only from the service name.
Instead of `resolveDynamicServiceAndMethodWithDotReplaced()`, should we write  `resolveDynamicService()` and `resolveDynamicServiceWithDotReplaced()`. These functions could be invoked from within `resolveDynamicServiceAndMethod` based on flag configuration.

Reply:
gRPC method name could not contain dots so it means same anyway.

Reply:
ah ok.

I saw `headers.setPath("/foo.bar.zoo/method.name?a=b");` in the tests that are being added. nvm